comp/haagent/impl/haagent.go

```diff
@@ -50,6 +50,15 @@ func (h *haAgentImpl) SetLeader(leaderAgentHostname string) {
 	h.isLeader.Store(agentHostname == leaderAgentHostname)
 }
 
+// ShouldRunIntegration return true if the agent integrations is should to run.
+// When ha-agent is disabled, the agent behave as standalone agent (non HA) and will always run all integrations.
+func (h *haAgentImpl) ShouldRunIntegration(integrationName string) bool {
+	if h.Enabled() {
+		return validHaIntegrations[integrationName] && h.isLeader.Load()
```

Comment:
I don't think this logic is correct: when I run the agent with `DD_HA_AGENT_ENABLED=true`, the agent skips all checks:
```
2024-11-27 16:26:36 UTC | CORE | DEBUG | (pkg/collector/worker/check_logger.go:76 in Debug) | check:uptime | Check is an HA integration and current agent is not leader, skipping execution...
2024-11-27 16:26:36 UTC | CORE | DEBUG | (pkg/collector/worker/check_logger.go:76 in Debug) | check:ntp | Check is an HA integration and current agent is not leader, skipping execution...
2024-11-27 16:26:36 UTC | CORE | DEBUG | (pkg/collector/worker/check_logger.go:76 in Debug) | check:service_discovery | Check is an HA integration and current agent is not leader, skipping execution...
2024-11-27 16:26:37 UTC | CORE | DEBUG | (pkg/collector/worker/check_logger.go:76 in Debug) | check:memory | Check is an HA integration and current agent is not leader, skipping execution...
2024-11-27 16:26:38 UTC | CORE | DEBUG | (pkg/collector/worker/check_logger.go:76 in Debug) | check:io | Check is an HA integration and current agent is not leader, skipping execution...
```

Could you also please add a test that makes sure that non-HA checks continue to run when HA-agent is enabled.

Reply:
@vickenty thanks for catching that, it's fixed here https://github.com/DataDog/datadog-agent/pull/31186/commits/f65d6d34acdfbf2a3c2e21091264c4b565d0e949 with updated tests.

looks like I introduced the bug when I moved move the logic from worker.go to haagent component: https://github.com/DataDog/datadog-agent/pull/31186/commits/5ed60bbfe99f9fa395e9f6cd87f830836a7d29a9#diff-fa837122b5056960e5f1bcbe5e6da5ed11701b8e75a2dff72681905d9a1e0314R225-R226